src/main/java/com/health/controller/HomeController.java

```diff
@@ -3890,6 +3895,206 @@ public String addTestimonialPost(HttpServletRequest req, Model model, Principal
+            spokenVideo.setSpokenVideoId(id);
+
+            try {
+
+                String folder = CommonData.uploadDirectorySpokenVideo + lanId;
+
+                String document = ServiceUtility.uploadMediaFile(file, env, folder);
+
+                if (spokenVideoService.findByFilePath(document) != null) {
+                    model.addAttribute("error_msg", "duplicate source location");
+                    return addSpokenVideoGet(req, principal, model);
+                }
+
+                spokenVideo.setFilePath(document);
+
+                spokenVideoService.save(spokenVideo);
+
+            } catch (Exception e) {
+
+                logger.error("Error in Add Spoken Video: {} {}", file, lanId, e);
+                model.addAttribute("error_msg", CommonData.RECORD_ERROR);
+                return addSpokenVideoGet(req, principal, model);
+            }
+        }
+
+        model.addAttribute("success_msg", CommonData.RECORD_SAVE_SUCCESS_MSG);
+        return addSpokenVideoGet(req, principal, model);
+
+    }
+
+    @GetMapping("/spokenVideo/edit/{id}")
+    public String editSpokenVideoGet(@PathVariable int id, HttpServletRequest req, Model model, Principal principal) {
+
+        User usr = getUser(principal);
+        logger.info("{} {} {}", usr.getUsername(), req.getMethod(), req.getRequestURI());
+
+        model.addAttribute("userInfo", usr);
+
+        SpokenVideo spokenVideo = spokenVideoService.findById(id);
+
+        if (spokenVideo == null) {
+
+            return "redirect:/addSpokenVideo";
+        }
+
+        model.addAttribute("spokenVideo", spokenVideo);
+
+        return "updateSpokenVideo";
+    }
+
+    @PostMapping("/updateSpokenVideo")
+    public String updateSpokenVideoPost(HttpServletRequest req, Model model, Principal principal,
+            @RequestParam("spokenVideoFile") MultipartFile file) {
+
+        User usr = getUser(principal);
+        logger.info("{} {} {}", usr.getUsername(), req.getMethod(), req.getRequestURI());
+
+        model.addAttribute("userInfo", usr);
+
+        String spokenVideoId = req.getParameter("spokenVideoId");
+        String displayName = req.getParameter("displayName");
+
+        SpokenVideo spokenVideo = spokenVideoService.findById(Integer.parseInt(spokenVideoId));
+
+        if (spokenVideo == null) {
+            model.addAttribute("error_msg", "Spoken Video doesn't exist");
+            return editSpokenVideoGet(Integer.parseInt(spokenVideoId), req, model, principal);
+        }
+
+        if (displayName == null || displayName.isEmpty()) {
+            model.addAttribute("error_msg", "display name should not be null");
+            return editSpokenVideoGet(Integer.parseInt(spokenVideoId), req, model, principal);
+        } else {
+            if (spokenVideoService.findByDisplayName(displayName) != null) {
+                int tempId = spokenVideoService.findByDisplayName(displayName).getSpokenVideoId();

```

Comment:
```suggestion
           SpokenVideo video = spokenVideoService.findByDisplayName(displayName);
            if (video != null) {
                int tempId = video.getSpokenVideoId();

```
Don't call the same function twice. Use a variable.